lib/Optimizer/Transforms/LoopAnalysis.cpp

```diff
@@ -327,6 +327,15 @@ bool opt::LoopComponents::isClosedIntervalForm() {
 
 bool opt::LoopComponents::isLinearExpr() { return addendValue || scaleValue; }
 
+template <typename T>
+constexpr int computeArgsOffset() {
+  if constexpr (std::is_same_v<T, cc::ConditionOp>) {
+    return 1;
+  } else {
+    return 0;
+  }
```

Comment:
This could probably be trimmed down to make it in accordance with LLVM style guide.
```suggestion
  if constexpr (std::is_same_v<T, cc::ConditionOp>)
    return 1;
  else
    return 0;
```
Or
```suggestion
  return constexpr (std::is_same_v<T, cc::ConditionOp>) ? 1 : 0;
```